### DISCUSSION TITLE
Timing information in the websocket text-to-speech API

### DISCUSSION THREAD

**Discussion by SoftMemes** [deepgram/community#844] [General help] [answered]
I've been exploring Deepgram for text-to-speech and while there appear to be no documentation of it on the website, the Python SDK has an implementation for this using websockets.

This would be perfect for our use case, we have streaming _input_ (from a language model) and want to stream audio output with as low latency as possible. However, we also need timing information, to correlate a piece of input text to the audio generated for it.

Other vendors like Elevenlabs provide this (in their case, timing down to word pieces). Are there any plans on either token level timing or  alternatively a way in the websocket connection to flag where the audio for one input chunk ends and the next one starts?

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by SandraRodgers** [comment, marked as answer]
The Python SDK text-to-speech websocket is in beta currently which is why you don't see documentation for it on the website, but we'll have docs out in the next couple weeks so check back for that. In the meantime you can read about the TTS rest API here: https://developers.deepgram.com/docs/python-sdk-text-to-speech

I have shared your suggestion with the product manager for TTS websockets. I'll report back later today if I find out any information that could be useful to you. This is particularly interesting to us right now while the API is in beta; thanks for the suggestion.

**Comment by michellelychan** [comment]
Hi @SoftMemes , I'm Michelle and I'm the product manager at Deepgram working on our websocket API for TTS. I'd love to learn more about your experience. Would you be free for a brief 1-1 call by any chance? If so, feel free to reach out to me on Discord and we can find a time. I'm michellelyc on Deepgram Discord : https://discord.gg/QSsy8frM 
Thank you!

**Comment by ANIL-KADURKA** [comment]
i just want to know webscokets for tts is it like takign less time than the rest api i just want to know  i have the url for the ur beta version i am trying to send the text to it but it is giving me the  data viaolitaion error i have treid using postman
